lib/control-connection.js

```diff
@@ -417,23 +418,21 @@ class ControlConnection extends events.EventEmitter {
 
   async _refreshControlConnection(hostIterator) {
 
-    if (this.options.sni) {
-      this.connection = this._borrowAConnection(hostIterator);
```

Comment:
This logic (an if/else case which sometimes catches errors and sometimes doesn't) was a complete mess.  In the Astra case we fell squarely into the SNI world but connection failures (which occur naturally when all nodes are killed) was causing errors to be thrown here, which in turn brought down the entire refresh job.  This has all been radically simplified in this PR.